### PR TITLE
'block_until' column added for restriction access before this date

### PR DESCRIPTION
**Pull request by Netmisa**
'block_until' field added. This field specify the end date restriction.

``` json
{
    "billing_plan": {
        "default": true,
        "max_request_count": null,
        "max_object_count": null,
        "id": 3,
        "name": "nav_ctp"
    },
    "authorizations": [],
    "keys": [],
    "end_point": {
        "default": true,
        "hostnames": [],
        "id": 1,
        "name": "navitia.io"
    },
    "login": "remy",
    "type": "with_free_instances",
    "id": 1,
    "block_until": "2015-12-24",
    "email": "remyabi@gmail.com"
}
```

### CODE REVIEW

File: source/tyr/migrations/versions/962cfcce76f_block_until_column_added_for_.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '962cfcce76f'
+down_revision = '11eae2b0e6b0'
+
+from alembic import op
+import sqlalchemy as sa
+
+def upgrade():
+    op.add_column('user', sa.Column('block_until', sa.Date(), nullable=True))
```
Comment (antoine-de) [inline source/tyr/migrations/versions/962cfcce76f_block_until_column_added_for_.py]: same as the other PR, why store it as a date ?

**Comment by antoine-de** [comment]
Have you checked the existing solution to manage the quota ? They don't seems enough for your needs ?

File: source/tyr/tyr/validations.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from datetime import datetime
+
+def datetime_format(value):
+    """Parse a valid looking date in the format YYYYmmddTHHmmss"""
+    utc_dt = datetime.strptime(value, "%Y%m%dT%H%M%S")
+    if utc_dt.year < 1900:
+        raise ValueError(u"Year must be >= 1900")
```
Comment (lrocheWB) [inline source/tyr/tyr/validations.py]: >= or > 1900 ?

Comment (lrocheWB) [inline source/tyr/tyr/validations.py]: For me, this verification is useless... In our case, we don't will need it.
it is normal "u" ?

File: source/tyr/migrations/versions/962cfcce76f_block_until_column_added_for_.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '962cfcce76f'
+down_revision = '11eae2b0e6b0'
+
+from alembic import op
+import sqlalchemy as sa
+
+def upgrade():
+    op.add_column('user', sa.Column('block_until', sa.DateTime(), nullable=True))
```
Comment (lrocheWB) [inline source/tyr/migrations/versions/962cfcce76f_block_until_column_added_for_.py]: Choose:
- In application, convert the time to UTC and stored using a TIMESTAMP WITH TIME ZONE column.
  In this case, add "timezone=True"
- Or set database's timezone to UTC.

http://stackoverflow.com/questions/6151084/which-timestamp-type-should-i-choose-in-a-postgresql-database

File: source/tyr/migrations/versions/2320ba18ce1_update_unlimited_billing_plans.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+revision = '2320ba18ce1'
+down_revision = '962cfcce76f'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.execute("update billing_plan set max_request_count=0 where max_request_count IS NULL")
+    op.execute("update billing_plan set max_object_count=0 where max_object_count IS NULL")
```
Comment (lrocheWB) [inline source/tyr/migrations/versions/2320ba18ce1_update_unlimited_billing_plans.py]: Function missing ;)

**Comment by lrocheWB** [comment]
Everythink is ok for me.

@pbougue, @kinnou02, @antoine-de can we merge this ? :)

File: source/tyr/tyr/resources.py [hunk 5]
```diff
@@ -123,15 +124,17 @@ def format(self, value):
...
     'end_point': fields.Nested(end_point_fields),
-    'billing_plan': fields.Nested(billing_plan_fields),
+    'billing_plan': fields.Nested(billing_plan_fields)
 }
 
 user_fields_full = {
     'id': fields.Raw,
     'login': fields.Raw,
     'email': fields.Raw,
+    'block_until': FieldDate,
```
Comment (antoine-de) [inline source/tyr/tyr/resources.py:137]: DateTime no ?

File: source/tyr/migrations/versions/962cfcce76f_block_until_column_added_for_.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '962cfcce76f'
+down_revision = '11eae2b0e6b0'
+
+from alembic import op
+import sqlalchemy as sa
+
+def upgrade():
+    op.add_column('user', sa.Column('block_until', sa.DateTime(timezone=True), nullable=True))
```
Comment (kinnou02) [inline source/tyr/migrations/versions/962cfcce76f_block_until_column_added_for_.py:17]: no timezone! all datetime in the database are stored in UTC

Comment (Netmisa) [inline source/tyr/migrations/versions/962cfcce76f_block_until_column_added_for_.py:17]: Yes, that we want

**Comment by Netmisa** [comment]
Thanks @kinnou02 
We change configuration of block_until column in tyr api. PR : https://github.com/CanalTP/navitia/pull/1346

**Pull request merged**